src/main/java/com/ahoo/issuetrackerserver/label/application/LabelService.java

```diff
@@ -0,0 +1,75 @@
+package com.ahoo.issuetrackerserver.label.application;
+
+import com.ahoo.issuetrackerserver.common.exception.DuplicatedLabelTitleException;
+import com.ahoo.issuetrackerserver.common.exception.ErrorMessage;
+import com.ahoo.issuetrackerserver.common.exception.NotExistsLabelException;
+import com.ahoo.issuetrackerserver.label.domain.Label;
+import com.ahoo.issuetrackerserver.label.domain.TextBrightness;
+import com.ahoo.issuetrackerserver.label.infrastructure.LabelRepository;
+import com.ahoo.issuetrackerserver.label.presentation.dto.LabelResponse;
+import com.ahoo.issuetrackerserver.label.presentation.dto.LabelUpdateRequest;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class LabelService {
+
+    private final LabelRepository labelRepository;
+
+    @Transactional(readOnly = true)
+    public List<LabelResponse> findAll() {
+        return labelRepository.findAll().stream()
+            .map(LabelResponse::from)
+            .collect(Collectors.toUnmodifiableList());
+    }
+
+    @Transactional
+    public void save(String title, String colorCode, String description, TextBrightness textBrightness) {
+        if (isExistsTitle(title)) {
+            throw new DuplicatedLabelTitleException(ErrorMessage.DUPLICATED_LABEL_TITLE);
+        }
+        labelRepository.save(Label.of(title, colorCode, description, textBrightness));
+    }
+
+    private boolean isExistsTitle(String title) {
+        return labelRepository.existsByTitle(title);
+    }
+
+    @Transactional
+    public void deleteById(Long id) {
+        try {
+            labelRepository.deleteById(id);
+        } catch (EmptyResultDataAccessException e) {
+            throw new NotExistsLabelException(ErrorMessage.NOT_EXISTS_LABEL);
+        }
+
+    }
+
+    @Transactional(readOnly = true)
+    public LabelResponse findById(Long id) {
+        Label findLabel = labelRepository.findById(id)
+            .orElseThrow(() -> new NotExistsLabelException(ErrorMessage.NOT_EXISTS_LABEL));
+
+        return LabelResponse.from(findLabel);
+    }
+
+    @Transactional
+    public void update(LabelUpdateRequest labelUpdateRequest, Long id) {
+        if (isExistsTitle(labelUpdateRequest.getTitle())) {
+            throw new DuplicatedLabelTitleException(ErrorMessage.DUPLICATED_LABEL_TITLE);
+        }
+
+        Label findLabel = labelRepository.findById(id)
+            .orElseThrow(() -> new NotExistsLabelException(ErrorMessage.NOT_EXISTS_LABEL));
```

Comment:
```suggestion
        Label findLabel = labelRepository.findById(id)
            .orElseThrow(() -> new NotExistsLabelException(ErrorMessage.NOT_EXISTS_LABEL));
            
        if (isExistsTitle(labelUpdateRequest.getTitle())) {
            throw new DuplicatedLabelTitleException(ErrorMessage.DUPLICATED_LABEL_TITLE);
        }
```

기존 로직을 고수하신다면, 순서를 바꿔서 id 먼저 검사하는게 아주 조금 성능이 더 낫지 않을까 싶습니다(비용이 덜드는 검증을 먼저 하기 때문에) 이것은 저만의 뇌피셜이기때문에,,, 아더의 의견도 궁금하네요,,,

Reply:
논리적으로 먼저 수행되야 하는 검증이 기존 리소스가 존재하는지 판단했어야 하는데 제가 깊게 생각하지 못했네요 ㅠㅠ
먼저 엔티티의 존재여부 먼저 확인하고 다음에 기존 타이틀과 중복되는지 확인하는 로직으로 변경하겠습니다 👍